File: notification-service/src/main/java/dev/zvaryyka/notificationservice/service/NotificationStatusService.java
package dev.zvaryyka.notificationservice.service;

import dev.zvaryyka.notificationservice.models.NotificationStatus;
import dev.zvaryyka.notificationservice.repository.NotificationStatusRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class NotificationStatusService {

    private final NotificationStatusRepository notificationStatusRepository;

    @Autowired
    public NotificationStatusService(NotificationStatusRepository notificationStatusRepository) {
        this.notificationStatusRepository = notificationStatusRepository;
    }
    public NotificationStatus findByStatusName(String statusName) {
        return notificationStatusRepository.findByStatusName(statusName);
    }
}

File: rebalancer-service/src/main/kotlin/dev/zvaryyka/rebalancerservice/service/NotificationStatusService.kt
package dev.zvaryyka.rebalancerservice.service

import dev.zvaryyka.rebalancerservice.models.NotificationStatus
import dev.zvaryyka.rebalancerservice.repository.NotificationStatusRepository
import org.springframework.stereotype.Service

@Service
class NotificationStatusService (
    private val NotificationStatusRepository: NotificationStatusRepository
){


    fun findByStatusName(toString: String): NotificationStatus? {
        return NotificationStatusRepository.findByStatusName(toString)
    }
}